src/main/java/in/projecteka/consentmanager/clients/model/PatientLinkReferenceResult.java

```diff
@@ -0,0 +1,38 @@
+package in.projecteka.consentmanager.clients.model;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import in.projecteka.consentmanager.link.discovery.model.patient.response.GatewayResponse;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.NotNull;
+import java.util.UUID;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(Include.NON_NULL)
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Data
+public class PatientLinkReferenceResult {
+    private UUID requestId;
+    private String timestamp;
+    private Link link;
+    private Error error;
+    @NotNull
+    private GatewayResponse resp;
+
+    public boolean hasResponseId(){
+        if (resp == null) {
+            return false;
+        }
+        if (resp.getRequestId() == null || "".equals(resp.getRequestId())) {
+            return false;
+        }
```

Comment:
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/StringUtils.html#StringUtils--

You can use this here.

Reply:
you can say 

return response != null && !StringUtils.isEmpty(resp.getRequestId());

Reply:
Done.